src/lib/CookieStorage.js

```diff
@@ -0,0 +1,312 @@
+        }
+    }
+
+    /**
+     * @param {string} name
+     * @param {string} [namespace]
+     * @returns {{metadata: CookieMetadata, data: Uint8Array} | null}
+     * @private
+     */
+    static _readCookies(name, namespace) {
+        const metadata = this._getMetaData(name, namespace);
+        if (!metadata) return null;
+
+        let cookieDataBase64Url = '';
+        for (let chunk = 0; chunk < metadata.chunkCount; ++chunk) {
+            const chunkData = CookieJar.readCookie(this._getChunkCookieName(name, chunk, namespace));
+            if (!chunkData) return null;
+            cookieDataBase64Url += chunkData;
+        }
+
+        const data = Nimiq.BufferUtils.fromBase64Url(cookieDataBase64Url)
+            .subarray(this._calculateEncodedMetadataSize(metadata));
+        return { data, metadata };
+    }
+
+    /**
+     * @param {string} name
+     * @param {string} [namespace]
+     * @returns {CookieMetadata | null}
+     * @private
+     */
+    static _getMetaData(name, namespace) {
+        const firstChunk = CookieJar.readCookie(this._getChunkCookieName(name, 0, namespace));
+        if (!firstChunk) return null;
+        // Read enough base64 data to include the longest possible metadata, if the cookie is long enough.
+        return this._decodeMetadata(Nimiq.BufferUtils.fromBase64Url(firstChunk.substring(0, 23)));
+    }
+
+    /* eslint-disable no-bitwise */
+    /**
+     * @param {Uint8Array} encodedMetadata
+     * @returns {CookieMetadata}
+     * @private
+     */
+    static _decodeMetadata(encodedMetadata) {
+        if (!encodedMetadata.byteLength) throw new Errors.KeyguardError('Invalid CookieStorage metadata.');
+        const isEncrypted = !!(encodedMetadata[0] & 0b1);
+        const chunkCount = encodedMetadata[0] >> 1;
+        /** @type {Uint8Array | undefined} */
+        let encryptionInitVector;
+
+        if (isEncrypted) {
+            encryptionInitVector = encodedMetadata.subarray(1, 1 + CookieStorage.ENCRYPTION_INIT_VECTOR_SIZE);
+            if (encryptionInitVector.byteLength !== CookieStorage.ENCRYPTION_INIT_VECTOR_SIZE) {
+                // not enough data provided
+                throw new Errors.KeyguardError('Invalid CookieStorage metadata.');
+            }
+        }
+
+        return {
+            isEncrypted,
+            chunkCount,
+            encryptionInitVector,
+        };
+    }
+
+    /**
+     * @param {CookieMetadata} metadata
+     * @returns {Uint8Array}
+     * @private
+     */
+    static _encodeMetadata(metadata) {
+        const encryptionInitVectorSize = metadata.encryptionInitVector ? metadata.encryptionInitVector.byteLength : 0;
+        if (metadata.chunkCount > 127
+            || (metadata.isEncrypted && !metadata.encryptionInitVector)
+            || (metadata.encryptionInitVector && encryptionInitVectorSize !== CookieStorage.ENCRYPTION_INIT_VECTOR_SIZE)
+        ) throw new Errors.KeyguardError('Invalid CookieStorage metadata.');
```

Comment:
Suggestion to put the first condition already on a new line, indented, to match with the other lines.

```suggestion
        if (
            metadata.chunkCount > 127
            || (metadata.isEncrypted && !metadata.encryptionInitVector)
            || (metadata.encryptionInitVector && encryptionInitVectorSize !== CookieStorage.ENCRYPTION_INIT_VECTOR_SIZE)
        ) throw new Errors.KeyguardError('Invalid CookieStorage metadata.');
```

Reply:
Nah, that doesn't look nice :D